### DISCUSSION TITLE
[feature request] Add shell completions?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [v2fly/v2ray-core#1958] [Ideas]
How about adding shell completions for bash/zsh/fish/posh/etc?

I guess <https://jmtirado.net/shell-completion-with-cobra-and-go/> and <https://github.com/cli/cli/blob/cf4b73ff958b272cf3c9c0cf9351459f76b793a0/pkg/cmd/completion/completion.go> may be helpful?

Thanks!

**Comment by kulongwangzhi85** [comment]
我更希望在dns解析阶段，能有fwmark功能添加或者iptables 或 nftables防火墙联动功能的添加。而不必到 出口阶段才打标记。
